src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -771,6 +775,19 @@ def _push(
         # pylint: disable=protected-access
         if isinstance(other, (FormulaEngine, FormulaEngine3Phase)):
             self._steps.append((TokenType.COMPONENT_METRIC, other))
+        elif isinstance(other, (Quantity, float)):
+            match oper:
+                case "+" | "-":
+                    if not isinstance(other, Quantity):
+                        raise RuntimeError(
+                            f"A Quantity must be provided for addition or subtraction to {other}"
+                        )
+                case "*" | "/":
+                    if not isinstance(other, float):
+                        raise RuntimeError(
+                            f"A float must be provided for scalar multiplication to {other}"
+                        )
```

Comment:
We need an extra check to deal with python weirdness (aka to be idiomatic)

```suggestion
                    if not isinstance(other, float) and not isinstance(other, int):
                        raise RuntimeError(
                            f"A float must be provided for scalar multiplication to {other}"
                        )

Reply:
fixed